osf/models/collection.py

```diff
@@ -107,13 +159,26 @@ def save(self, *args, **kwargs):
             if self.title != 'Bookmarks':
                 # Bookmark collections are always named 'Bookmarks'
                 self.title = 'Bookmarks'
+        saved_fields = self.get_dirty_fields() or []
         ret = super(Collection, self).save(*args, **kwargs)
+
         if first_save:
             # Set defaults for M2M
             self.collected_types = ContentType.objects.filter(app_label='osf', model__in=['abstractnode', 'collection'])
             # Set up initial permissions
             self.update_group_permissions()
             self.get_group('admin').user_set.add(self.creator)
+
+        elif 'is_public' in saved_fields:
+            cgms = list(self.collectedguidmetadata_set.all())
+
+            if self.is_public:
+                # Add all collection submissions back to ES index
+                self.bulk_update_search(cgms)
```

Comment:
This should probably happen in celery (queued to run after the request), like we do for nodes. Otherwise, failures to update ES will result in the whole transaction being rolled back 😱